### ISSUE TITLE
add a warning when people put comments above the shebang

### ISSUE THREAD

**Issue by vapier** [koalaman/shellcheck#844]
Sometimes people don't realize the shebang must be the first line and think it's OK to put comments before it.  Especially when they test it out by running `./foo.sh` and it still works (due to default script behavior in the kernel exec paths).  It would be nice if we warned about this explicitly.

By default, `SC2148` will fire with a warning like:
`Tips depend on target shell and yours is unknown. Add a shebang.`

But if you're already passing `--shell=bash`, that warning is omitted, and we get no warnings at all.

#### Here's a snippet or screenshot that shows the problem:

```sh
# Some comment
#!/bin/bash
```

To avoid false positives, i think we should warn only if comments precede the shebang.  i.e. this one should not warn:
```sh
# Some comment
some_code=''
#!/bin/bash
```

#### Here's what I wanted or expected to see:

`Shebang must be the first line`

**Comment by VincentClaverie** [comment]
+1

**Comment by koalaman** [comment]
This has been added as described. It was definitely not an uncommon problem, and it is indeed easy to miss in a review or during test. Thanks!
